### PR TITLE
Card search using the Search Bar

### PR DESCRIPTION
**Pull request by acnormun**
### Type of Change
* * [ ] Bugfix
* * [x] Feature
* * [ ] Code style update (formatting, local variables)
* * [ ] Refactoring (no functional changes, no api changes)
* * [ ] Tests
* * [ ] Other

### Motivation and Context
Add feature - Search Bar.

### Summary of Changes
Added a new component for searching words in send cards and their unit tests.

### Demonstration 
image

### CODE REVIEW

File: src/components/search/Search.module.scss
Comment (Aldemylla) [inline src/components/search/Search.module.scss:1]: Some pixels can still be replaced by unnnic tokens here

File: src/components/search/Search.test.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  describe('closeSearch()', () => {
+    it('should call closeSearch', () => {
+      const { instance } = setup(true);
+      const spy = spyOn('closeSearch');
+      expect(spy).not.toHaveBeenCalled();
+      instance.closeSearch();
+      expect(spy).toHaveBeenCalledTimes(1);
+    });
+  });
+});
```
Comment (Aldemylla) [inline src/components/search/Search.test.tsx:260]: Adding spaces between `describe` functions will make the code a little more readable:

```suggestion
describe(SearchBar.name, () => {
  it('should render searchbar', () => {
    const { baseElement } = render(<SearchBar {...baseProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  describe('handleInput()', () => {
    it('should call handleInput', () => {
      const { getByPlaceholderText } = render(<SearchBar {...baseProps} />);
      const input = getByPlaceholderText('...');
      expect(input).toBeTruthy();
    });
  });

  describe('getAllNodes()', () => {
    it('should call getAllNodes', () => {
      const { instance } = setup(true);
      const spy = spyOn('getAllNodes');
      expect(spy).not.toHaveBeenCalled();
      instance.getAllNodes();
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should return filtered nodes', () => {
      const { instance } = setup(true);
      const response = instance.getAllNodes();
      expect(response).toEqual(filteredNode);
    });
  });

  describe('findNodes()', () => {
    it('should call findNodes', () => {
      const { instance } = setup(true);
      const spy = spyOn('findNodes');
      expect(spy).not.toHaveBeenCalled();
      instance.findNodes('teste');
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should return found nodes', () => {
      const { instance } = setup(true);
      const response = instance.findNodes('teste');
      expect(response).toEqual(filteredNode);
    });
  });

  describe('dragBackground()', () => {
    it('should call dragBackground', () => {
      const { instance } = setup(true);
      const spy = spyOn('dragBackground');
      expect(spy).not.toHaveBeenCalled();
      instance.dragBackground();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });

  describe('toggleMoveSelected()', () => {
    it('should call toggleMoveSelected', () => {
      const { instance } = setup(true);
      const spy = spyOn('toggleMoveSelected');
      expect(spy).not.toHaveBeenCalled();
      instance.toggleMoveSelected('up');
      expect(spy).toHaveBeenCalledTimes(1);
    });
    it('should call dragBackground', () => {
      const { instance } = setup(true);
      const spy = spyOn('dragBackground');
      expect(spy).toHaveBeenCalledTimes(2);
      instance.toggleMoveSelected('up');
      expect(spy).toHaveBeenCalledTimes(3);
    });
    it('should set search', () => {
      const { instance, props } = setup(true);
      instance.toggleMoveSelected('down');
      expect(props.search).not.toEqual({});
    });
  });

  describe('closeSearch()', () => {
    it('should call closeSearch', () => {
      const { instance } = setup(true);
      const spy = spyOn('closeSearch');
      expect(spy).not.toHaveBeenCalled();
      instance.closeSearch();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
```

File: src/components/search/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      } else {
+        return false;
+      }
+    });
+  }
+
+  private dragBackground() {
+    const canvasBg = document.getElementById('panzoom');
+    if (this.props.search.nodes[this.props.search.selected]) {
+      const ui = this.props.search.nodes[this.props.search.selected].data.ui.position;
```
Comment (Aldemylla) [inline src/components/search/SearchBar.tsx:79]: Destructuring `this.props.search` will result in fewer repetitions, which in addition to making the code more readable, will make maintenance easier:

```suggestion
    const { nodes, selected } = this.props.search
    if (nodes[selected]) {
      const ui = nodes[selected].data.ui.position;
```

File: src/components/search/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+      const width = window.innerWidth / 2;
+      const height = window.innerHeight / 2;
+      if (canvasBg) {
+        canvasBg.style.transform = `matrix(1, 0, 0, 1, ${width - ui.left}, ${height - ui.top})`;
+      }
+    }
+  }
+
+  private toggleMoveSelected(type: 'up' | 'down') {
```
Comment (Aldemylla) [inline src/components/search/SearchBar.tsx:88]: You could destructure `this.props.search` in this function too

File: src/components/search/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+  }
+
+  private closeSearch() {
+    this.props.handleSearchChange({
+      active: false,
+      value: this.props.search.value,
+      nodes: this.props.search.nodes,
+      selected: 0
+    });
+  }
```
Comment (Aldemylla) [inline src/components/search/SearchBar.tsx:124]: You could destructure this.props.search in this function too:
```suggestion
  private closeSearch() {
    const { value, nodes } = this.props.search;
  
    this.props.handleSearchChange({
      active: false,
      value,
      nodes,
      selected: 0
    });
  }
```

File: src/components/search/SearchBar.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        </span>
+        <Button
+          name={''}
+          onClick={() => this.closeSearch()}
+          type={ButtonTypes.ghost}
+          iconName="close-1"
+          size="small"
+          leftSpacing={true}
+        />
+      </div>
```
Comment (Aldemylla) [inline src/components/search/SearchBar.tsx:161]: Semantics is missing, could you add it?

File: src/components/sidebar/Sidebar.tsx [hunk 6]
```diff
@@ -221,27 +235,48 @@ export class Sidebar extends React.PureComponent<SidebarStoreProps, {}> {
...
+        >
+          <UnnnicTooltip
+            className={styles.left_aligned}
+            text={'Search'}
+            side="right"
+            shortcutText={this.props.selectionActive ? 'Ctrl C' : null}
+          >
+            <div className={styles.option} onClick={() => this.handleSearchChanged()}>
+              <span className="material-symbols-rounded">search</span>
+            </div>
```
Comment (Aldemylla) [inline src/components/sidebar/Sidebar.tsx:255]: Semantics is missing here too

File: src/flowTypes.ts [hunk 7]
```diff
@@ -299,8 +299,10 @@ export interface Flow {
 }
 
 export interface Action {
+  [x: string]: any;
```
Comment (Aldemylla) [inline src/flowTypes.ts:302]: What is the meaning of "`x`", is this the most descriptive name that could be used here?

**Review by Aldemylla — CHANGES_REQUESTED** [review]

File: src/components/canvas/Canvas.module.scss [hunk 8]
```diff
@@ -11,16 +11,18 @@ $background_offset: ($grid_size / 2) + ($node_padding) + px;
...
-  background-color: $unnnic-color-background-snow;
   background-position: 0px 0px;
   background-image: radial-gradient(
     circle at center,
-    $unnnic-color-neutral-soft 0,
-    $unnnic-color-neutral-soft 1px,
-    $unnnic-color-background-snow 2px,
-    $unnnic-color-background-snow 100%);
+    #D0D3D9 0,
+    #D0D3D9 1px,
```
Comment (Aldemylla) [inline src/components/canvas/Canvas.module.scss:18]:
```suggestion
    $unnnic-color-neutral-cleanest 0,
    $unnnic-color-neutral-cleanest 1px,
```

File: src/components/search/components/CloseButton.module.scss [hunk 9]
```diff
@@ -0,0 +1,13 @@
+.closeButton{
+    display: flex;
+    height: 100%;
+    justify-content: center;
+    align-content: center;
+
+    .closeIcon{
+        justify-items: center;
+        align-items: center;
+        padding: 4px;
```
Comment (Aldemylla) [inline src/components/search/components/CloseButton.module.scss:10]:
```suggestion
        padding: $unnnic-spacing-nano;
```

File: src/components/sidebar/components/SearchButton.module.scss [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+.searchButton{
+    display: flex;
+    height: 100%;
+    width: 100%;
+    justify-content: center;
+    align-content: center;
+
+    .searchIcon{
+        border-radius: 4px;
```
Comment (Aldemylla) [inline src/components/sidebar/components/SearchButton.module.scss:12]:
```suggestion
        border-radius: $unnnic-border-radius-sm;
```

**Review by Aldemylla — CHANGES_REQUESTED** [review]

**Review by paulobernardoaf — CHANGES_REQUESTED** [review]

**Review by paulobernardoaf — APPROVED** [review]

**Review by Aldemylla — APPROVED** [review]

**Pull request merged**